ReleaseHistory.md

```diff
@@ -1,4 +1,7 @@
 # SARIF Package Release History (SDK, Driver, Converters, and Multitool)
+## UNRELEASED
+* PRF: Switch file system traversal to pre-order with producer-consumer to accelerate time to scan first artifact.
```

Comment:
>Switch [](http://example.com/codeflow?start=7&length=6)

I feel like we will require some testing for this. Did you look at the existing tests for the ordered file specifier class?

Reply:
Just dug into the existing coverage a bit.  We do have functional coverage.  Does stuff like checking that we're hitting the right number of files.  I also ran coverage.  [1]  I'm not sure why I'm showing line coverage instead of block, (and need to be shifting focus presently) but it's about 85% from the Sarif.Driver tests.  Running more of our existing tests probably won't get this number up any because based on code highlighting, pretty much all of the missed lines are in error paths.

Those are in two categories: 1) the CheckFaulted() return statements on the worker thread, and 2) the Exception handler in EnumeratedArtifacts() that deals with Exceptions coming out of the Channel and cancellation.

If I Moq CheckFaulted, I probably get it over 90% but maybe fall a bit short of 95%.  I don't think it makes sense to try to hit every early-return path out of the worker thread.

[1] 
![image](https://github.com/microsoft/sarif-sdk/assets/145607305/9f51f43a-0289-4d15-9dcd-8a5e4a758fe8)